### ISSUE TITLE
Issues with system-wide installation

### ISSUE THREAD

**Issue by TheChymera** [datalad/datalad-fuse#116]
I was able to get it to work via `pip install`, but for some reason, even though the tests seem to pass, the system-wide install I set up doesn't work.

Some of the tests are skipped, specifically those referring to “fuse” so I assume maybe that's indicative of the issue, maybe it's somehow not linking to fuse itself?
Any ideas what could be wrong?

This is the issue
```console
[deco]/mnt/data/datalad ❱ datalad fusefs -d dandisets /tmp/dandisets-fuse
[ERROR  ] No module named 'fuse'
```

But datalad wtf lists fuse:

```console
[deco]/mnt/data/datalad ❱ datalad wtf | ag fuse
  - fuse:
    - description: DataLad FUSE command suite
      - datalad_fuse.FuseFS:
        - class: FuseFS
        - module: datalad_fuse
          - fusefs
          - fusefs
      - datalad_fuse.fsspec_cache_clear.FsspecCacheClear:
        - module: datalad_fuse.fsspec_cache_clear
      - datalad_fuse.fsspec_head.FsspecHead:
        - module: datalad_fuse.fsspec_head
    - module: datalad_fuse
```

<details>

```console
[deco]~ ❱ datalad wtf
# WTF
## configuration <SENSITIVE, report disabled by configuration>
## credentials
  - keyring:
    - active_backends:
      - EncryptedKeyring with [PBKDF2] AES256.CFB v.1.0 at /home/chymera/.local/share/python_keyring/crypted_pass.cfg
    - config_file: /home/chymera/.config/python_keyring/keyringrc.cfg
    - data_root: /home/chymera/.local/share/python_keyring
## datalad
  - version: 1.0.2
## dependencies
  - annexremote: 1.6.0
  - cmd:7z: 17.05
  - cmd:annex: 10.20230926
  - cmd:bundled-git: UNKNOWN
  - cmd:git: 2.44.0
  - cmd:ssh: 9.7p1
  - cmd:system-git: 2.44.0
  - cmd:system-ssh: 9.7p1
  - humanize: 4.9.0
  - iso8601: 2.1.0
  - keyring: 25.1.0
  - keyrings.alt: 5.0.1
  - msgpack: 1.0.8
  - platformdirs: 4.2.0
  - requests: 2.31.0
## environment
  - LANG: en_GB.utf8
  - PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/bin:/usr/lib/llvm/18/bin:/usr/lib/llvm/17/bin
## extensions
  - fuse:
    - description: DataLad FUSE command suite
    - entrypoints:
      - datalad_fuse.FuseFS:
        - class: FuseFS
        - module: datalad_fuse
        - names:
          - fusefs
          - fusefs
      - datalad_fuse.fsspec_cache_clear.FsspecCacheClear:
        - class: FsspecCacheClear
        - module: datalad_fuse.fsspec_cache_clear
        - names:
          - fsspec-cache-clear
          - fsspec_cache_clear
      - datalad_fuse.fsspec_head.FsspecHead:
        - class: FsspecHead
        - module: datalad_fuse.fsspec_head
        - names:
          - fsspec-head
          - fsspec_head
    - module: datalad_fuse
    - version: 0.5.1
## git-annex
  - build flags:
    - Assistant
    - Webapp
    - Pairing
    - Inotify
    - DBus
    - DesktopNotify
    - TorrentParser
    - MagicMime
    - Benchmark
    - Feeds
    - Testsuite
    - S3
    - WebDAV
  - dependency versions:
    - aws-0.24.2
    - bloomfilter-2.0.1.0
    - crypton-0.34
    - DAV-1.3.4
    - feed-1.3.2.1
    - ghc-9.2.8
    - http-client-0.7.16
    - persistent-sqlite-2.13.3.0
    - torrent-10000.1.1
    - uuid-1.3.15
    - yesod-1.6.2.1
  - key/value backends:
    - SHA256E
    - SHA256
    - SHA512E
    - SHA512
    - SHA224E
    - SHA224
    - SHA384E
    - SHA384
    - SHA3_256E
    - SHA3_256
    - SHA3_512E
    - SHA3_512
    - SHA3_224E
    - SHA3_224
    - SHA3_384E
    - SHA3_384
    - SKEIN256E
    - SKEIN256
    - SKEIN512E
    - SKEIN512
    - BLAKE2B256E
    - BLAKE2B256
    - BLAKE2B512E
    - BLAKE2B512
    - BLAKE2B160E
    - BLAKE2B160
    - BLAKE2B224E
    - BLAKE2B224
    - BLAKE2B384E
    - BLAKE2B384
    - BLAKE2BP512E
    - BLAKE2BP512
    - BLAKE2S256E
    - BLAKE2S256
    - BLAKE2S160E
    - BLAKE2S160
    - BLAKE2S224E
    - BLAKE2S224
    - BLAKE2SP256E
    - BLAKE2SP256
    - BLAKE2SP224E
    - BLAKE2SP224
    - SHA1E
    - SHA1
    - MD5E
    - MD5
    - WORM
    - URL
    - X*
  - operating system: linux x86_64
  - remote types:
    - git
    - gcrypt
    - p2p
    - S3
    - bup
    - directory
    - rsync
    - web
    - bittorrent
    - webdav
    - adb
    - tahoe
    - glacier
    - ddar
    - git-lfs
    - httpalso
    - borg
    - hook
    - external
  - supported repository versions:
    - 8
    - 9
    - 10
  - upgrade supported from repository versions:
    - 0
    - 1
    - 2
    - 3
    - 4
    - 5
    - 6
    - 7
    - 8
    - 9
    - 10
  - version: 10.20230926
## location
  - path: /home/chymera
  - type: directory
## metadata.extractors
## metadata.filters
## metadata.indexers
## python
  - implementation: CPython
  - version: 3.11.8
## system
  - distribution: gentoo/2.15/n/a
  - encoding:
    - default: utf-8
    - filesystem: utf-8
    - locale.prefered: UTF-8
  - filesystem:
    - CWD:
      - max_pathlength: 4096
      - mount_opts: rw,noatime
      - path: /home/chymera
      - type: ext4
    - HOME:
      - max_pathlength: 4096
      - mount_opts: rw,noatime
      - path: /home/chymera
      - type: ext4
    - TMP:
      - max_pathlength: 4096
      - mount_opts: rw,noatime
      - path: /tmp
      - type: ext4
  - max_path_length: 269
  - name: Linux
  - release: 6.7.4-gentoo
  - type: posix
  - version: #1 SMP PREEMPT_DYNAMIC Tue Feb 13 20:12:30 EST 2024
```

</details>

Full build and test log → https://ppb.chymera.eu/f9fac2.html

Files installed by the extension → https://ppb.chymera.eu/fa6aca.log

**Comment by yarikoptic** [comment]
given the number of similar reports -- in case of "no module" errors, please first  analyze `setup.{py,cfg}` for a list of dependencies to add to your gentoo packaging.  Here in particular I bet we are getting `fuse` module via 

`fsspec[fuse,http] >= 2022.1.0, != 2022.10.0`

specification.  Hence -- check which `fuse` library fsspec depends on and add it directly as dependency

**Comment by TheChymera** [comment]
Admittedly on occasion I miss things, but yes, I always consult the python files for deps.

Both `dev-python/fsspec` and `sys-fs/fuse` are pulled in.
The one thing I notices while I was setting this up last week but forgot when I filed this issue, is that I noticed the flags for fsspec, which the Gentoo package does not provide.
I thought it would mean that the package just ships all features, but maybe it doesn't.

These are the packages that fsspec depends on:

<details>

```console
[deco]~ ❱ equery g fsspec
 * Searching for fsspec ...

 * dependency graph for dev-python/fsspec-2024.3.1
 `--  dev-python/fsspec-2024.3.1  amd64
   `--  dev-lang/python-3.10.14_p1  (dev-lang/python) amd64
   `--  dev-lang/python-3.11.9  (dev-lang/python) amd64
   `--  dev-lang/python-3.12.3  (dev-lang/python) amd64
   `--  dev-python/aiohttp-3.9.5  (dev-python/aiohttp) amd64  [python_targets_python3_10(-)? python_targets_python3_11(-)? python_targets_python3_12(-)?]
   `--  dev-python/numpy-1.26.4  (dev-python/numpy) ~amd64  [python_targets_python3_10(-)? python_targets_python3_11(-)? python_targets_python3_12(-)?]
   `--  dev-python/pytest-asyncio-0.23.6  (dev-python/pytest-asyncio) amd64  [python_targets_python3_10(-)? python_targets_python3_11(-)? python_targets_python3_12(-)?]
   `--  dev-python/pytest-mock-3.14.0  (dev-python/pytest-mock) ~amd64  [python_targets_python3_10(-)? python_targets_python3_11(-)? python_targets_python3_12(-)?]
   `--  dev-python/requests-2.31.0  (dev-python/requests) amd64  [python_targets_python3_10(-)? python_targets_python3_11(-)? python_targets_python3_12(-)?]
   `--  dev-python/tqdm-4.66.4  (dev-python/tqdm) ~amd64  [python_targets_python3_10(-)? python_targets_python3_11(-)? python_targets_python3_12(-)?]
   `--  dev-vcs/git-2.45.0  (dev-vcs/git) ~amd64
   `--  dev-python/pytest-8.2.0  (>=dev-python/pytest-7.4.4) ~amd64  [python_targets_python3_10(-)? python_targets_python3_11(-)? python_targets_python3_12(-)?]
   `--  dev-python/gpep517-15  (>=dev-python/gpep517-15) ~amd64  [python_targets_python3_10(-)? python_targets_python3_11(-)? python_targets_python3_12(-)?]
   `--  dev-python/setuptools-69.5.1-r1  (>=dev-python/setuptools-69.0.3) amd64  [python_targets_python3_10(-)? python_targets_python3_11(-)? python_targets_python3_12(-)?]
[ dev-python/fsspec-2024.3.1 stats: packages (14), max depth (1) ]
```

</details>

The files list of the installed package shows that indeed some fuse files are installed, but maybe this is insufficient?

```console
[deco]~ ❱ equery f fsspec | ag fuse
/usr/lib/python3.11/site-packages/fsspec/__pycache__/fuse.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/fuse.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/fuse.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/fuse.py
```

<details>

```console
[deco]~ ❱ equery f fsspec
 * Searching for fsspec ...
 * Contents of dev-python/fsspec-2024.3.1:
/usr
/usr/lib
/usr/lib/python3.11
/usr/lib/python3.11/site-packages
/usr/lib/python3.11/site-packages/fsspec
/usr/lib/python3.11/site-packages/fsspec-2024.3.1.dist-info
/usr/lib/python3.11/site-packages/fsspec-2024.3.1.dist-info/METADATA
/usr/lib/python3.11/site-packages/fsspec-2024.3.1.dist-info/WHEEL
/usr/lib/python3.11/site-packages/fsspec-2024.3.1.dist-info/top_level.txt
/usr/lib/python3.11/site-packages/fsspec/__init__.py
/usr/lib/python3.11/site-packages/fsspec/__pycache__
/usr/lib/python3.11/site-packages/fsspec/__pycache__/__init__.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/__init__.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/__init__.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/_version.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/_version.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/_version.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/archive.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/archive.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/archive.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/asyn.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/asyn.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/asyn.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/caching.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/caching.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/caching.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/callbacks.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/callbacks.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/callbacks.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/compression.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/compression.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/compression.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/config.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/config.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/config.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/conftest.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/conftest.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/conftest.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/core.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/core.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/core.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/dircache.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/dircache.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/dircache.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/exceptions.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/exceptions.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/exceptions.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/fuse.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/fuse.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/fuse.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/generic.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/generic.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/generic.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/gui.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/gui.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/gui.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/mapping.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/mapping.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/mapping.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/parquet.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/parquet.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/parquet.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/registry.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/registry.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/registry.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/spec.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/spec.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/spec.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/transaction.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/transaction.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/transaction.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/utils.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/utils.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/__pycache__/utils.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/_version.py
/usr/lib/python3.11/site-packages/fsspec/archive.py
/usr/lib/python3.11/site-packages/fsspec/asyn.py
/usr/lib/python3.11/site-packages/fsspec/caching.py
/usr/lib/python3.11/site-packages/fsspec/callbacks.py
/usr/lib/python3.11/site-packages/fsspec/compression.py
/usr/lib/python3.11/site-packages/fsspec/config.py
/usr/lib/python3.11/site-packages/fsspec/conftest.py
/usr/lib/python3.11/site-packages/fsspec/core.py
/usr/lib/python3.11/site-packages/fsspec/dircache.py
/usr/lib/python3.11/site-packages/fsspec/exceptions.py
/usr/lib/python3.11/site-packages/fsspec/fuse.py
/usr/lib/python3.11/site-packages/fsspec/generic.py
/usr/lib/python3.11/site-packages/fsspec/gui.py
/usr/lib/python3.11/site-packages/fsspec/implementations
/usr/lib/python3.11/site-packages/fsspec/implementations/__init__.py
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/__init__.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/__init__.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/__init__.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/arrow.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/arrow.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/arrow.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/cache_mapper.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/cache_mapper.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/cache_mapper.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/cache_metadata.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/cache_metadata.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/cache_metadata.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/cached.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/cached.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/cached.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/dask.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/dask.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/dask.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/data.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/data.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/data.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/dbfs.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/dbfs.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/dbfs.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/dirfs.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/dirfs.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/dirfs.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/ftp.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/ftp.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/ftp.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/git.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/git.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/git.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/github.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/github.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/github.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/http.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/http.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/http.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/jupyter.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/jupyter.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/jupyter.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/libarchive.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/libarchive.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/libarchive.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/local.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/local.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/local.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/memory.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/memory.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/memory.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/reference.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/reference.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/reference.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/sftp.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/sftp.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/sftp.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/smb.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/smb.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/smb.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/tar.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/tar.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/tar.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/webhdfs.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/webhdfs.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/webhdfs.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/zip.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/zip.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/__pycache__/zip.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/implementations/arrow.py
/usr/lib/python3.11/site-packages/fsspec/implementations/cache_mapper.py
/usr/lib/python3.11/site-packages/fsspec/implementations/cache_metadata.py
/usr/lib/python3.11/site-packages/fsspec/implementations/cached.py
/usr/lib/python3.11/site-packages/fsspec/implementations/dask.py
/usr/lib/python3.11/site-packages/fsspec/implementations/data.py
/usr/lib/python3.11/site-packages/fsspec/implementations/dbfs.py
/usr/lib/python3.11/site-packages/fsspec/implementations/dirfs.py
/usr/lib/python3.11/site-packages/fsspec/implementations/ftp.py
/usr/lib/python3.11/site-packages/fsspec/implementations/git.py
/usr/lib/python3.11/site-packages/fsspec/implementations/github.py
/usr/lib/python3.11/site-packages/fsspec/implementations/http.py
/usr/lib/python3.11/site-packages/fsspec/implementations/jupyter.py
/usr/lib/python3.11/site-packages/fsspec/implementations/libarchive.py
/usr/lib/python3.11/site-packages/fsspec/implementations/local.py
/usr/lib/python3.11/site-packages/fsspec/implementations/memory.py
/usr/lib/python3.11/site-packages/fsspec/implementations/reference.py
/usr/lib/python3.11/site-packages/fsspec/implementations/sftp.py
/usr/lib/python3.11/site-packages/fsspec/implementations/smb.py
/usr/lib/python3.11/site-packages/fsspec/implementations/tar.py
/usr/lib/python3.11/site-packages/fsspec/implementations/webhdfs.py
/usr/lib/python3.11/site-packages/fsspec/implementations/zip.py
/usr/lib/python3.11/site-packages/fsspec/mapping.py
/usr/lib/python3.11/site-packages/fsspec/parquet.py
/usr/lib/python3.11/site-packages/fsspec/registry.py
/usr/lib/python3.11/site-packages/fsspec/spec.py
/usr/lib/python3.11/site-packages/fsspec/tests
/usr/lib/python3.11/site-packages/fsspec/tests/abstract
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__init__.py
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/__init__.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/__init__.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/__init__.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/common.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/common.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/common.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/copy.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/copy.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/copy.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/get.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/get.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/get.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/put.cpython-311.opt-1.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/put.cpython-311.opt-2.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/__pycache__/put.cpython-311.pyc
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/common.py
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/copy.py
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/get.py
/usr/lib/python3.11/site-packages/fsspec/tests/abstract/put.py
/usr/lib/python3.11/site-packages/fsspec/transaction.py
/usr/lib/python3.11/site-packages/fsspec/utils.py
/usr/share
/usr/share/doc
/usr/share/doc/fsspec-2024.3.1
/usr/share/doc/fsspec-2024.3.1/README.md.bz2
```

</details>

**Comment by yarikoptic** [comment]
I guess `sys-fs/fuse` is fuse library, not some python fuse bindings...



> fsspec[fuse,http] >= 2022.1.0, != 2022.10.0

means that it requests `fsspec` installation with `optional_requires` for `fuse`, which are defined in either `setup.cfg` or `setup.py` or `pyproject.toml`. In their case apparently in the latter: https://github.com/fsspec/filesystem_spec/blob/master/pyproject.toml where they have

```
[project.optional-dependencies]
...
fuse = ["fusepy"]
```

so what you need is to add gentoo package for `fusepy`.

To a degree I would consider it a bug in our packaging here since we do now import `fuse` directly so we better announce it directly as well -- will do now in
- https://github.com/datalad/datalad-fuse/pull/117

**Comment by TheChymera** [comment]
Ok, I think I got confused by the required package because there are a number of packages called python and fuse:

* https://github.com/libfuse/pyfuse3/
* https://github.com/libfuse/python-fuse

I got the correct one, actually called fusepy now, and it works so the issue is done.

However I notice that the latest release is 6 years old and I couldn't find any test suite I could run for the package → https://github.com/fusepy/fusepy 
Might be worth considering regarding long-term sustainability. Seems to work with contemporary python but that might change.
